diff --git a/photon_rs.d.ts b/photon_rs.d.ts
index fc477c6dd3416d3d7873711159e1650f82333e19..f0f65446a180ce36f53722a177523120da21f580 100644
--- a/photon_rs.d.ts
+++ b/photon_rs.d.ts
@@ -1,3 +1,5 @@
+export function setWasm(wasm: any): void;
+
 /* tslint:disable */
 /* eslint-disable */
 /**
diff --git a/photon_rs_bg.js b/photon_rs_bg.js
index 6279c156155076480155191f4d0afdaeb2d472a5..0471116d19af08f2b9b0c89d2506f5f08ffa6678 100644
--- a/photon_rs_bg.js
+++ b/photon_rs_bg.js
@@ -1,4 +1,8 @@
-import * as wasm from './photon_rs_bg.wasm';
+let wasm;
+
+export const setWasm = (w) => {
+    wasm = w;
+}
 
 const heap = new Array(32).fill(undefined);
 
